Add index.md (#1850)
canonical/repo-policy-compliance: charm/docs/index.md

Like any Juju charm, this charm supports one-line deployment, configuration, integration, scaling, and more. For `repo-policy-compliance`, this includes:
* Customising enabled policies 
* Running in debug mode
* Choosing different GitHub authentication methods
* Modifying Flask-specific features like a secret key for security-related needs, the run environment (e.g., production) or where the application is mounted

See the [Actions](https://charmhub.io/repo-policy-compliance/actions) and [Configurations](https://charmhub.io/repo-policy-compliance/configurations) tabs to learn more about the actions and configurations supported by this charm.

## Get started <a name="get-started"></a>
### What you'll need <a name="what-youll-need"></a>
* A Kubernetes cloud.
* Juju 3 installed and a controller created.

> * NGINX Ingress Controller. If you’re using MicroK8s, this can be done by running the command `microk8s enable ingress`. 

Reviewer: ```suggestion
```

I think Ingress is not required for the purpose of the tutorial.

```
Model        Controller  Cloud/Region        Version  SLA          Timestamp
repo-policy  microk8s    microk8s/localhost  3.1.8    unsupported  12:21:02+02:00

App             Version  Status  Scale  Charm                   Channel      Rev  Address         Exposed  Message
postgresql-k8s  14.12    active      1  postgresql-k8s          14/edge      272  10.152.183.64   no       Primary
repo-policy              active      1  repo-policy-compliance  latest/edge   66  10.152.183.139  no       

Unit               Workload  Agent  Address      Ports  Message
postgresql-k8s/0*  active    idle   10.1.72.161         Primary
repo-policy/0*     active    idle   10.1.72.167           
```

### Generate an authentication token and test <a name="token"></a>

Generate a one-time authentication token for `repo-policy-compliance` using `curl` and save it as the `ONE_TIME_TOKEN` environment variable. You will need the IP address of the Unit for the `repo-policy-compliance` charm; in the example output above, 10.1.72.167 is the necessary IP address. You will also need the token used for `charm_token` configuration (in this example, "abc"). 

```

> ONE_TIME_TOKEN = curl http://10.1.72.167:8000/one-time-token -H"Authorization: Bearer abc" 

Reviewer: ```suggestion
ONE_TIME_TOKEN=$(curl http://10.1.72.167:8000/one-time-token -H"Authorization: Bearer abc" )
```

Author: Thanks for catching!

```

The variable should look similar to the following:
```
d156dda1f03df9d42fd788d93799c57b4275ca5facccce92ef9b91cf4fc13f6a%
```

Finally, use the `repo-policy-compliance` Unit IP address and the one-time token to check for compliance on a GitHub repository formatted as `OWNER/REPO`. 

```
curl -i http://10.1.72.167:8000/push/check-run -H"Authorization: Bearer $ONE_TIME_TOKEN" --data '{"repository_name": "OWNER/REPO"}' -H"Content-Type: application/json"
```

The output should look similar to the following:

```
HTTP/1.1 204 NO CONTENT
Server: gunicorn
Date: Wed, 11 Sep 2024 12:09:11 GMT
Connection: close
Content-Type: text/html; charset=utf-8
```

> "NO CONTENT" indicates that the repository is compliant.

Reviewer: ```suggestion
The `204` status code indicates that the repository is compliant.
```
